src/libraries/System.Linq.Expressions/tests/Block/BlockTests.cs

```diff
@@ -249,5 +249,104 @@ .maxstack 0
   .maxcontinuation 0
 }");
         }
+
+        [Theory]
+        [ClassData(typeof(CompilationTypes))]
+        public static void MutableValue(bool useInterpreter)
+        {
+            // f = () =>
+            //     {
+            //         ValueTypeWithParameterlessConstructorThatThrows p;
```

Comment:
I feel like I might not understand the semantics of this interpreter. Do we just implicitly initialize locals to `default`? Do you think it would improve clarity to throw in an `= default` here?

Reply:
Yes, we implicitly initialize the value to `default` in `MutableValue`. I was not including `= default` to ensure we caught regressions where the behavior of `MutableValue` changes. Since we're using a value type that throws an exception in the parameterless constructor, that might also catch the same regressions though.